app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb

```diff
@@ -129,67 +129,15 @@ def available_vlans_and_hosts(options = {})
     # Remove certain networks
     vlans.delete_if { |_k, v| v.in?(['Service Console', 'VMkernel']) }
 
-    unless @vlan_options[:dvs] == false
-      rails_logger('allowed_dvs', 0)
-      vlans_dvs = allowed_dvs(@vlan_options, hosts)
-      vlans.merge!(vlans_dvs)
-      rails_logger('allowed_dvs', 1)
-    end
-
-    return vlans, hosts
-  end
-
-  def allowed_dvs(_options = {}, hosts = nil)
-    @dvs_ems_connect_ok ||= {}
-    @dvs_by_host ||= {}
-    switches = {}
-    src = get_source_and_targets
-    return switches if src.blank?
-
-    hosts ||= get_selected_hosts(src)
-
-    # Find if we need to connect to the EMS to collect a host's dvs
-    missing_hosts = hosts.reject { |h| @dvs_by_host.key?(h.id) }
-    unless missing_hosts.blank?
-      begin
-        st = Time.now
-        return switches if src[:ems] && @dvs_ems_connect_ok[src[:ems].id] == false
-        vim = load_ar_obj(src[:ems]).connect
-        missing_hosts.each { |dest_host| @dvs_by_host[dest_host.id] = get_host_dvs(dest_host, vim) }
-      rescue
-        @dvs_ems_connect_ok[src[:ems].id] = false if src[:ems]
-        return switches
-      ensure
-        vim.disconnect if vim rescue nil
-        _log.info "Network DVS collection completed in [#{Time.now - st}] seconds"
+    # dvportgroups key-value transformed
+    if options[:dvs]
+      dvlans = Hash.new { |h, k| h[k] = [] }
+      hosts.to_miq_a.each { |h| h.lans.each { |l| dvlans[l.name] << l.switch.name if l.switch.shared } }
```

Comment:
The `l.switch.shared` should use the predicate method `l.switch.shared?`.

I'd also like to see the outer block use the `do ... end` version because I think it makes it more obvious that it's a nested loop.